### DISCUSSION TITLE
Vercel Design Site Navbar

### DISCUSSION THREAD

**Discussion by tonydattolo** [vercel/next.js#46563] [Help]
Was wondering if you could share how you built the dynamic navbar on the vercel design site https://vercel.com/design

How do you calculate the start location of the top of the component, and how do you calculate when it has reached 16px from the top?

![Screenshot from 2023-02-28 10-52-52](https://user-images.githubusercontent.com/46014916/221907564-3251a333-0103-4c93-a29c-dea6c895d5d5.png)
![Screenshot from 2023-02-28 10-53-00](https://user-images.githubusercontent.com/46014916/221907567-06833a1b-a7a8-486d-a8d4-87327a40613c.png)

**Comment by ciruz** [comment]
Depending on your design framework there is maybe a solution out of the box.

https://getbootstrap.com/docs/5.1/components/scrollspy/

But it's pretty simple https://dev.to/areeburrub/change-nav-link-s-style-as-you-scroll-4p62
